build-time-analytics/src/test/java/io/quarkus/ts/buildtimeanalytics/AbstractAnalyticsIT.java

```diff
@@ -0,0 +1,160 @@
+        assertTrue(remoteConfigFile.exists());
+        JsonPath json = JsonPath.from(remoteConfigFile);
+        assertFalse(json.getList("deny_anonymous_ids", String.class).contains("fake-anonymous-id"));
+    }
+
+    protected void verifyBuildSuccessful(Result buildResult) {
+        assertTrue(buildResult.isSuccessful(), "The application didn't build on JVM. Output: " + buildResult.getOutput());
+    }
+
+    protected void verifyBuildPropertyRecognized(Result buildResult, String property) {
+        assertFalse(buildResult.getOutput().contains(String.format(UNRECOGNIZED_PROPERTY_FORMAT, property)));
+    }
+
+    protected void verifyPayloadAbsent(QuarkusCliRestService app) {
+        File payload = getAnalyticsPayloadFile(app);
+        assertFalse(payload.exists());
+    }
+
+    protected void verifyValidPayloadPresent(QuarkusCliRestService app) {
+        File payload = getAnalyticsPayloadFile(app);
+        assertTrue(payload.exists());
+        validatePayload(app, payload);
+    }
+
+    private void validatePayload(QuarkusCliRestService app, File payload) {
+        JsonPath json = JsonPath.from(payload);
+        validateJsonValues(json);
+        validateExtensions(app, json);
+    }
+
+    private void validateJsonValues(JsonPath json) {
+        Map<String, String> values = Map.of(
+                "event", getEventType(),
+                "context.ip", QUARKUS_ANALYTICS_IP,
+                "context.quarkus.version", QuarkusProperties.getVersion(),
+                "context.java.vendor", System.getProperty("java.vendor"),
+                "context.java.version", System.getProperty("java.version"),
+                "context.os.name", System.getProperty("os.name"),
+                "context.os.os_arch", System.getProperty("os.arch"),
+                "context.os.version", System.getProperty("os.version"));
+        values.forEach((key, value) -> assertEquals(value, json.getString(key)));
+    }
+
+    private void validateExtensions(QuarkusCliRestService app, JsonPath json) {
+        List<Dependency> pomDependencies = getPomDependencies(app);
+        List<String> pomDependencyGAs = pomDependencies.stream()
+                .map(dependency -> mapToGA(dependency.getGroupId(), dependency.getArtifactId()))
+                .collect(Collectors.toList());
+        List<PayloadExtension> payloadExtensions = json.getList("properties.app_extensions", PayloadExtension.class);
+        List<String> payloadExtensionGAs = payloadExtensions.stream()
+                .map(payloadExtension -> mapToGA(payloadExtension.getGroupId(), payloadExtension.getArtifactId()))
+                .collect(Collectors.toList());
+        assertEquals(pomDependencyGAs, payloadExtensionGAs);
+        payloadExtensions.stream()
+                .map(PayloadExtension::getVersion)
+                .sorted()
+                .distinct()
+                .forEach(version -> Matchers.matchesPattern(getExtensionVersionPattern()));
```

Comment:
If I understand this correctly, these lines:
1. Read versions
2. Sort the versions
3. Drop duplicates
4. Ignore the results (but create the same new matcher for each of them)

Reply:
Yes, it's a silly code, thanks for spotting it.